equivariant_attention/modules.py

```diff
@@ -342,6 +342,52 @@ def forward(self, features, **kwargs):
         return output
 
 
+#####################################
+class G1x1SE3_q(nn.Module):
+    """Graph Linear SE(3)-equivariant layer, equivalent to a 1x1 convolution.
+
+    This is equivalent to a self-interaction layer in TensorField Networks.
+    """
+    def __init__(self, f_in, f_out, fv_in, fv_out, learnable=True):
+        """SE(3)-equivariant 1x1 convolution.
+
+        Args:
+            f_in: input Fiber() of feature multiplicities and types
+            f_out: output Fiber() of feature multiplicities and types
+        """
+        super().__init__()
+        self.f_in = f_in
+        self.f_out = f_out
+        self.fv_in = fv_in
+        self.fv_out = fv_out
+
+        # Linear mappings: 1 per output feature type
+        self.transform = nn.ParameterDict()
+        for m_out, d_out in self.f_out.structure:
+            for mv_in, dv_in in self.fv_in.structure:
+                for mv_out, dv_out in self.fv_out.structure:
+                    m_in = self.f_in.structure_dict[d_out]
+                    self.transform[f'({d_out},{dv_in},{dv_out})'] = nn.Parameter(torch.randn(m_out, m_in) / np.sqrt(m_in), requires_grad=learnable)
+
+    def __repr__(self):
+         return f"G1x1SE3(structure={self.f_out})"
+
+    def forward(self, features, **kwargs):
+        output = {}
+        tr_keys = []
+        for i in self.transform.keys():
```

Comment:
I think we might have an error here, #`self.transform.keys()` is equal to #`({d_out},{dv_in},{dv_out})`. It is not equal to #`d_out`s.

Reply:
I think it's good. The tr_keys only contains d_out via int(eval(i)[0]). And fiber object doesnt have degrees (as from file fibers.py)

Reply:
I tested. f_out.degrees and tr_keys contain the same elements.